linkerd/retry/src/lib.rs

```diff
@@ -0,0 +1,162 @@
+#![deny(warnings, rust_2018_idioms)]
+#![forbid(unsafe_code)]
+#![allow(clippy::inconsistent_struct_constructor)]
+#![allow(clippy::type_complexity)]
```

Comment:
Appears to be unneeded

```suggestion
#![allow(clippy::type_complexity)]
```